lib/client-api/src/story_store.ts

```diff
@@ -375,7 +376,9 @@ export default class StoryStore {
     const { passArgsFirst = true } = combinedParameters;
     const __isArgsStory = passArgsFirst && original.length > 0;
 
-    this._argTypesEnhancers.push(inferArgTypes); // lowest priority
+    // run at the end
+    this._argTypesEnhancers.push(inferArgTypes);
+    this._argTypesEnhancers.push(inferControls);
```

Comment:
@shilman shouldn't we reset it somewhere? Right now it leads to applying the same enhancers repeatedly, because they're remaining from all the previously processed stories. So if you have 100 stories, there is 100xinferArgTypes +  100xinferControls when processing the last of them